### PR TITLE
Changes to show int test logs

### LINKED ISSUE #36: bug: Test logs are not printing to std ouput
Currently, our logs from the integration tests are not showing in stdout while running `cargo test`. 

Fix it.

### PR DESCRIPTION
**Pull request by Shourya742**
This PR closes #36 #86

### CODE REVIEW

File: src/test_framework.rs [hunk 1]
```diff
@@ -188,10 +188,6 @@ impl TestFramework {
         *self.shutdown.write().unwrap() = true;
         // stop bitcoind
         let _ = self.bitcoind.client.stop().unwrap();
-        // Remove test temp dir, ignore error.
-        if fs::remove_dir_all::<PathBuf>(self.temp_dir.clone()).is_err() {
-            // Do Nothing
-        }
```
Comment (rajarshimaitra) [inline src/test_framework.rs:194]: Why? This will pollute the test datadir with run files. 

Maybe take this out as an independent `clean()` method? Then, it can be selectively commented out in tests to persist files if needed for a test.

but this seems unrelated to the logging problem. So maybe in a separate PR?

Comment (rajarshimaitra) [reply src/test_framework.rs:194]: Yeah I think this should be the right behavior. What we have currently is wrong. `stop` and `clear` should not be done in a single method.

Comment (rajarshimaitra) [reply src/test_framework.rs:194]: wrote it in an issue.
https://github.com/utxo-teleport/teleport-transactions/issues/86

**Review by rajarshimaitra — COMMENTED** [review]
ACK. Logs are showing up in CI now. 

But it still doesn't show the logs form the tests itself. Like this line. 

https://github.com/utxo-teleport/teleport-transactions/blob/9ec7d4547be2a7afe42032fa13bddee11c6b89ca/tests/abort2_case1.rs#L34

but this PR is good to go, modulo the above cleanup comment. And the missing logs from test situation could be documented in another new issue.

File: src/test_framework.rs (hunk 1 above)
Comment (wthrajat) [reply src/test_framework.rs:194]: What about making temporary directory under `/tmp`?
Something like `/tmp/teleport/tests`. Then we don't have to worry about cleaning as they'll get purged on every reboot.
But I can see some downsides to it, such as what if someone do not want to wait until next reboot and is testing again and again. This will probably just clutter the space.

And this will require a similar approach for others OSes like Windows for example as `/tmp` is standard for only Unix based OSes.

Comment (Shourya742) [reply src/test_framework.rs:194]: @rajarshimaitra we did discussed this in discord. But ya this clean and stop method also works.

Comment (rajarshimaitra) [reply src/test_framework.rs:194]: > What about making temporary directory under /tmp?

Thats more cleaner. I don't remember why I haven't done that. Worth a PR.

Comment (rajarshimaitra) [reply src/test_framework.rs:194]: > @rajarshimaitra we did discussed this in discord. But ya this clean and stop method also works.

I forgot what was the discussion. At first glance, I don't see why we would want to remove this line.

**Review by rajarshimaitra — COMMENTED** [review]
ACK 651a52b8796afec9d6d8dce798b7681bd4f73875

modulo https://github.com/utxo-teleport/teleport-transactions/pull/85/files#r1457841670

File: README.md [hunk 2]
```diff
@@ -63,16 +63,22 @@ The project includes both unit and integration tests. The integration tests simu
...
 
 ```console
+# without logs
 $ cargo test
+
+# with logs
+$ cargo test -- --nocapture
 ```
 
 To run the integration tests, `--features integration-test` flag must be enabled. Run integration tests with:
```
Comment (rajarshimaitra) [inline README.md:73]: As we are adding the feature into default list so we don't need this line anymore.

File: Cargo.toml [hunk 3]
```diff
@@ -24,7 +24,7 @@ bitcoind = {version = "0.32", features = ["25_0"] }
 
 #Empty default feature set, (helpful to generalise in github actions)
 [features]
-default = []
+default = ["integration-test"]
```
Comment (rajarshimaitra) [inline Cargo.toml:27]: It seems to me we don't need the `integration-test` feature at all. As usual I forgot why I added it. probably some dependency requirements that got solved afterward.

File: README.md [hunk 4]
```diff
@@ -63,16 +63,22 @@ The project includes both unit and integration tests. The integration tests simu
...
 
 ```console
+# without logs
 $ cargo test --features integration-test
-```
 
-To print out logs on the tests, set the `RUST_LOG` env variable to either `info`, `warn` or `error`.
+# with logs
+$ cargo test --features integration-test -- --nocapture
+```
```
Comment (rajarshimaitra) [inline README.md:81]: lets only keep `cargo test -- --nocapture`.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
ACK.

lets try to remove the `integration-test` feature here. We don't need it anymore and its not the standard practice.

**Review by wthrajat — APPROVED** [review]

**Review by rajarshimaitra — APPROVED** [review]
ACK 29bd5457bd8531fbcbb09506a120b0f0e306027f
